lib/redis_client/ruby_connection.rb

```diff
@@ -98,21 +100,22 @@ def write_timeout=(timeout)
     end
 
     def write(command)
-      buffer = RESP3.dump(command)
+      @buffer.clear
+      RESP3.dump(command, @buffer)
```

Comment:
So `String#clear` here free the `malloced` region: https://bugs.ruby-lang.org/issues/17790.

So this save allocating a string slot, but will `malloc` anyway which will eventually trigger a GC.

If we want to be smart we want to re-used that malloced region, but have to be careful not to end up with a giant buffer, so we'd need to clear it if it past a certain size. Unfortunately the only way to get the size of the `malloc` is via `ObjectSpace.memsize_of(str)`. I'd need to check its performance first.

We also have to be super careful not to leak data, as that happened in the past with a similar optimization https://github.com/ruby/net-protocol/pull/19

Reply:
> If we want to be smart we want to re-used that malloced region

Can we currently do this from ruby? I see the proposed patch into ruby is not merged yet.

Reply:
> Can we currently do this from ruby?

Well, if you pass the string to `read` without clearing it, it will re-use that space.

Reply:
But we are using it just for write in this case (`write` and `write_multi`), so this is not yet possible?

Reply:
Ah yeah, sorry I missed we where passing it to `RESP3.dump`. Yeah I see no solution here.

I think if we call `clear` we start over from an empty string, so we lose the pre-allocation benefits.